engine/src/main/java/com/odysseusinc/arachne/executionengine/service/impl/DescriptorServiceImpl.java

```diff
@@ -47,6 +48,9 @@ public class DescriptorServiceImpl implements DescriptorService {
     // TODO Consider replacing this flag with a dedicated logger
     private final boolean dependencyMatching;
 
+    @Getter
+    private volatile List<Descriptor> descriptors = Collections.emptyList();
```

Comment:
It looks like that descriptor contains only tarball descriptors. Can you somehow reflect that in the name?

Reply:
This should be reflected in the name of the service. This change is due, however I have consciously chosed to postpone this renaming to avoid increasing the amount of changes in this PR.